File: morse-code/morse_code.rb
#utf-8

# To run just execute:
# ruby morse_code.rb

class MorseCode
	attr_reader :plan_text

	def initialize(plan_text = 'Hello World')
	  @plan_text = plan_text.downcase!
	end
	
	def convert
		
		keys = { 'a' => '.-', 'b' => '-...', 'c' => '-.-.', 'd' => '-..', 'e' => '.', 'f' => '..-.', 
						 'g' => '--.', 'h' => '....', 'i' => '..', 'j' => '.---',	'k' => '-.-', 'l' => '.-..', 
						 'm' => '--', 'n' => '-.', 'o' => '---', 'p' => '.--.',	'q' => '--.-', 'r' => '.-.',
						 's' => '...', 't' => '-', 'u' => '..-', 'v' => '...-',	'w' => '.--', 'x' => '-..-',
						 'y' => '-.--', 'z' => '--..'
		}
		
		keys.each do |k, v|
			@plan_text.gsub!(k, "#{v} ")
	  end
	  
		@plan_text
	end
end

puts "What did you want to transform to morse code? "
word = gets.chomp

puts "YOUR WORD IN PLAIN TEXT:      #{word}"
c = MorseCode.new word
c.convert
puts "YOUR WORD IN MORSE CODE:      #{c.plan_text}"

File: morse-code/MorseCode.java
import java.util.Scanner;
import java.util.HashMap;
import java.util.Map;
 
public class MorseCode {

	private static String plainText;
	
  public static void main(String[] args) {
 
    Scanner scan = new Scanner(System.in);
    
	  System.out.println("What did you want to transform to morse code? ");
    plainText = scan.nextLine();
 
    scan.close();
    
    System.out.println(plainText);
  }
  
  public static String convert(){
  	HashMap<String, String> map = new HashMap<String, String>();
		
		map.put("a", ".-");
		map.put("b", "-...");
		map.put("c", "-.-.");
		map.put("d", "-..");
		map.put("e", ".");
		map.put("f", "..-.");
		map.put("g", "--.");
		map.put("h", "....");
		map.put("i", "..");
		map.put("j", ".---");
		map.put("k", "-.-");
		map.put("l", ".-..");
		map.put("m", "--");
		map.put("n", "-.");
		map.put("o", "---");
		map.put("p", ".--.");
		map.put("q", "--.-");
		map.put("r", ".-.");
		map.put("s", "...");
		map.put("t", "-");
		map.put("u", "..-");
		map.put("v", "...-");
		map.put("w", ".--");
		map.put("x", "-..-");
		map.put("y", "-.--");
		map.put("z", "--..");
						
  	return "";
  }
}
